GVFS/GVFS.Upgrader/GVFS.Upgrader.csproj

```diff
@@ -1,137 +1,63 @@
-      <HintPath>..\..\..\packages\NuGet.Protocol.4.9.2\lib\net46\NuGet.Protocol.dll</HintPath>
-    </Reference>
-    <Reference Include="NuGet.Versioning, Version=4.9.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\NuGet.Versioning.4.9.2\lib\net46\NuGet.Versioning.dll</HintPath>
-    </Reference>
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="System.IdentityModel" />
-    <Reference Include="System.IO.Compression, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089, processorArchitecture=MSIL" />
-    <Reference Include="System.Net.Http.WebRequest" />
-    <Reference Include="System.Runtime.Serialization" />
-    <Reference Include="System.Security" />
-    <Reference Include="System.ServiceModel" />
-    <Reference Include="System.Xml.Linq" />
-    <Reference Include="System.Data.DataSetExtensions" />
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System.Data" />
-    <Reference Include="System.Net.Http" />
-    <Reference Include="System.Xml" />
-  </ItemGroup>
-  <ItemGroup>
-    <Compile Include="..\GVFS.PlatformLoader\PlatformLoader.Windows.cs">
-      <Link>PlatformLoader.Windows.cs</Link>
-    </Compile>
-    <Compile Include="UpgradeOrchestrator.cs" />
-    <Compile Include="Program.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="App.config" />
-    <None Include="packages.config">
-      <SubType>Designer</SubType>
-    </None>
-  </ItemGroup>
-  <ItemGroup>
-    <BootstrapperPackage Include=".NETFramework,Version=v4.6.1">
-      <Visible>False</Visible>
-      <ProductName>Microsoft .NET Framework 4.6.1 %28x86 and x64%29</ProductName>
-      <Install>true</Install>
-    </BootstrapperPackage>
-    <BootstrapperPackage Include="Microsoft.Net.Framework.3.5.SP1">
-      <Visible>False</Visible>
-      <ProductName>.NET Framework 3.5 SP1</ProductName>
-      <Install>false</Install>
-    </BootstrapperPackage>
+    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
+    <PackageReference Include="CommandLineParser" Version="2.1.1-beta" />
   </ItemGroup>
+
   <ItemGroup>
-    <Analyzer Include="..\..\..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
-    <Analyzer Include="..\..\..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
+    <ProjectReference Include="..\GVFS.Common\GVFS.Common.csproj" />
+    <ProjectReference Include="..\GVFS.Virtualization\GVFS.Virtualization.csproj" /> 
   </ItemGroup>
+
+  <!-- ItemGroup Conditions are not supported on VS for Mac and Choose/When must be
+       Used instead, see https://github.com/mono/monodevelop/issues/7417 -->
+  <Choose>
+    <When Condition="'$(OS)' == 'Windows_NT'">

```

Comment:
The existing pattern to have separate `csproj` for Mac and Windows (like in `GVFS.Mount` project) helps avoid the need to do platform checks like this. But your approach of keeping a single project file for both platforms looks cleaner. Also now we end up having 2 different patterns (existing pattern of separate platform specific `csproj` files and this new common `csproj` with platform specific source code inclusions.)
 
 It would be good if we could discuss and decide on a single pattern that we could follow consistently.  @nickgra, @wilbaker, @kewillford, @jrbriggs - thoughts?

Reply:
If we can have a single file for both targets, I think that is preferable in the long run. If we can make it work now, I view this as a step in the right direction. At this point, I don't see any blockers in having everything as a single file (debugging on VS for Mac works, I will look again and make sure everything works on VS for Windows).

If there is a problem with using a single file, then my plan was to fall back to having separate projects.

Reply:
> It would be good if we could discuss and decide on a single pattern that we could follow consistently. @nickgra, @wilbaker, @kewillford, @jrbriggs - thoughts?

When @nickgra updated FastFetch to be cross-platform he was able to use with a single csproj (although I believe in that case it's .NET Core on both platforms).  

I believe that historically we couldn't do this because the ProjFS nuget package was incompatible with `PackageReference`.  Now that the layout has been fixed we should be able to start switching to using single projects (see also #94).

cc: @nickgra for his thoughts

Reply:
I think long term we want to move everything to .NET Core and moving to a single project is a good step in that direction. IIRC, `PackageReference` was what was stopping us from doing this. 

If this works fine on VSMac, I'm happy with this solution for now, although it does have the con of making us unable to build Mac on Windows as William called out in another comment.